lib/Dialect/XTenNN/IR/XTenNNOps.cpp

```diff
@@ -215,15 +223,90 @@ static void printKernelArgumentList(OpAsmPrinter &p, TypeRange types,
 }
 
 // Parse
-//  $name custom<KernelArgumentList>(type($arguments), $arguments) attr-dict
+// {((name = )?value, )*((name = )?value)}
+static ParseResult parseKernelInstantiationArgs(OpAsmParser &p,
+                                                SmallVector<Attribute> &values,
+                                                SmallVector<Attribute> &names) {
+  if (failed(p.parseLBrace()))
+    return failure();
+
+  if (failed(p.parseCommaSeparatedList([&p, &names, &values]() {
+        std::string name;
+        bool hasName = false;
+        if (succeeded(p.parseOptionalKeywordOrString(&name))) {
+          hasName = true;
+          if (failed(p.parseEqual()))
+            return failure();
+        }
+        Attribute attr;
+        auto res = p.parseOptionalAttribute(attr);
+        if (res.has_value() && succeeded(*res)) {
+          if (hasName)
+            names.push_back(StringAttr::get(p.getContext(), name));
```

Comment:
Could we do this push_back at the point where we currently do `hasName = true;`?